File: scraper/arbitrasje/sources/elkjop.py
"""Kampanjevarer fra elkjop.no.

IKKE IMPLEMENTERT: nettverkspolicyen i utviklingsmiljøet blokkerer
elkjop.no, så API-et kunne ikke inspiseres. elkjop.no bruker et internt
GraphQL-/søkelag. Fremgangsmåte: åpne en kampanjeside med Network-fanen
åpen, finn XHR/GraphQL-kallet som leverer produktlisten, og implementer på
samme mønster som power.py (egen parse-funksjon som kan testes med
fixtures). Ikke parse HTML — API-et finnes.
"""

from ..models import ShopOffer

name = "elkjop"


def fetch_offers(http) -> list[ShopOffer]:
    raise NotImplementedError("elkjop: endepunkt ikke kartlagt ennå (se kommentar i elkjop.py)")

File: scraper/src/adapters/elkjop.ts
import type { MarketplaceAdapter, RawListing } from "../types.ts";

/**
 * Nypris fra elkjop.no.
 *
 * IKKE IMPLEMENTERT: nettverkspolicyen i utviklingsmiljøet blokkerer
 * elkjop.no, så søke-API-et kunne ikke undersøkes. elkjop.no bruker et
 * internt GraphQL-/søkelag — åpne Network-fanen i nettleseren under et søk
 * for å finne endepunktet, og implementer på samme mønster som power-
 * adapteren (fetchJson + egen parse-funksjon som kan testes med fixtures).
 */
export const elkjopAdapter: MarketplaceAdapter = {
  source: "elkjop",
  async search(_query: string): Promise<RawListing[]> {
    throw new Error("elkjop-adapteren er ikke implementert ennå (se kommentar i elkjop.ts)");
  },
};
